lib/charms/magma_orc8r_libs/v0/orc8r_base.py

```diff
@@ -231,10 +226,79 @@ def _relations_ready(self) -> bool:
             return False
         return True
 
+    @property
+    def _pebble_layer(self) -> Layer:
+        """Returns pebble layer for the charm.
+
+        Returns:
+            Layer: Pebble Layer
+        """
+        return Layer(
+            {
+                "summary": f"{self.service_name} layer",
+                "description": f"pebble config layer for {self.service_name}",
+                "services": {
+                    self.service_name: {
+                        "override": "replace",
+                        "summary": self.service_name,
+                        "startup": "enabled",
+                        "command": self.startup_command,
+                        "environment": self._environment_variables,
+                    }
+                },
+            }
+        )
+
+    @property
+    def _environment_variables(self) -> dict:
+        """A set of environment variables required by the workload service.
+
+        Returns:
+            dict: Required environment variables
+        """
+        environment_variables = {}
+        default_environment_variables = {
+            "SERVICE_HOSTNAME": self.container_name,
+            "SERVICE_REGISTRY_MODE": "k8s",
+            "SERVICE_REGISTRY_NAMESPACE": self.namespace,
+        }
+        environment_variables.update(self.additional_environment_variables)
```

Comment:
Nitpicking, we can use the union operator for merging dictionaries. Like this 
```suggestion
        environment_variables |= self.additional_environment_variables
```
https://peps.python.org/pep-0584/

Reply:
I think `.update` is way easier to understand :)

Reply:
Yeah, I agree. It is in PEP so it was worth mentioning.